itinerant-iterators/tests/test_steganography.py

```diff
@@ -0,0 +1,15 @@
+# flake8: noqa
+import tempfile
+from pathlib import Path
+
+from app.steganography import Lsb, decode, encode
+
+
+def test_lsb_encode_decode():
+    message = "secret message"
+    lsb = Lsb()
+    f = tempfile.NamedTemporaryFile("w", delete=True, suffix=".png")
+    encode(message, Path("tests/data/python-logo.jpg"), Path(f.name), lsb)
+    got = decode(Path(f.name), lsb)
+    f.close()
+    assert message == got
```

Comment:
Solid integration test. Could be made cleaner though.
```suggestion
def test_lsb_encode_decode():
    message = "secret message"
    with tempfile.NamedTemporaryFile("w", delete=True, suffix=".png") as _f:
        temporary_file = Path(_f)
        encode(message, Path("tests/data/python-logo.jpg"), temporary_file, algorithm=Lsb())
        decoded_message = decode(temporary_file, lsb)
    assert message == decoded_message
```

If you don't understand what the `with` statement is, [here's a Real Python link](https://realpython.com/python-with-statement/), but anyway, it just manages the cleanup of the temporary file. Once execution leaves the with block, the temporary file is closed (and deleted).